ddtrace/contrib/django/patch.py

```diff
@@ -189,13 +189,20 @@ def traced_cache(django, pin, func, instance, args, kwargs):
                 db.ROWCOUNT, sum(1 for doc in result if doc) if result and isinstance(result, Iterable) else 0
             )
         elif command_name == "get":
-            # if valid result and check for special case for Django~3.0 that returns an empty Sentinel object as
-            # missing key
-            if result is not None and result != getattr(instance, "_missing_key", None):
-                span.set_metric(db.ROWCOUNT, 1)
-            # else result is invalid or None, set row count to 0
-            else:
+            try:
+                # if valid result and check for special case for Django~3.0 that returns an empty Sentinel object as
+                # missing key
+                if hasattr(result, "empty"):
```

Comment:
This I believe is internal knowledge of `pandas`. I would be inclined to use this approach as a last resort. I think it might be more beneficial if we could understand exactly why we still have an issue here and try to address that in its most general form. Otherwise we would end up having to do these sorts of lookups for every framework that exposes objects with a similar behaviour as `DataFrame`. WDYT?

Reply:
I've been trying to think about this but can't come up with what the more "general form" approach would be here, although I agree that doing this for every object that has the same behavior as `DataFrame` doesn't seem ideal.

Reply:
Good point. I changed this approach to use the id() function since None is a Python built-in singleton and any value set to None will also return the same id, so that should be safe. For the dataframe bug, checking if the result is an iterable first before using `==` fixed the ambiguity problem.